apps/examples/wifi_manager_sample/wm_test.c

```diff
@@ -0,0 +1,661 @@
+
+#define SET_SOFTAP_OPTION(opt, func, ssid, pwd)	\
+	do {										\
+		opt->func = func;						\
+		opt->ssid = ssid;						\
+		opt->password = pwd;					\
+	} while (0);
+
+#define SET_STA_OPTION(opt, func, ssid, auth, crypto, pwd)	\
+	do {													\
+		opt->func = func;									\
+		opt->ssid = ssid;									\
+		opt->auth_type = auth;								\
+		opt->crypto_type = crypto;							\
+		opt->password = pwd;								\
+	} while (0);
+
+#define WM_TEST_TIME_DIFF												\
+	(g_end_time.tv_sec - g_st_time.tv_sec)*1000 + (g_end_time.tv_usec - g_st_time.tv_usec)/1000
+
+#define WM_TEST_SIGNAL										\
+	do {													\
+		pthread_mutex_lock(&g_wm_mutex);					\
+		printf("T%d send signal\n", getpid());	\
+		pthread_cond_signal(&g_wm_cond);					\
+		pthread_mutex_unlock(&g_wm_mutex);					\
+	} while (0)
+
+#define WM_TEST_WAIT								\
+	do {											\
+		pthread_mutex_lock(&g_wm_mutex);			\
+		printf(" T%d wait signal\n", getpid());		\
+		pthread_cond_wait(&g_wm_cond, &g_wm_mutex);	\
+		pthread_mutex_unlock(&g_wm_mutex);			\
+	} while (0)
+
+#define WM_TEST_FUNC_SIGNAL								\
+	do {												\
+		pthread_mutex_lock(&g_wm_func_mutex);			\
+		printf(" T%d send func signal\n", getpid());	\
+		pthread_cond_signal(&g_wm_func_cond);			\
+		pthread_mutex_unlock(&g_wm_func_mutex);			\
+	} while (0)
+
+#define WM_TEST_FUNC_WAIT										\
+	do {														\
+		pthread_mutex_lock(&g_wm_func_mutex);					\
+		printf(" T%d wait func signal\n", getpid());			\
+		pthread_cond_wait(&g_wm_func_cond, &g_wm_func_mutex);	\
+		pthread_mutex_unlock(&g_wm_func_mutex);					\
+	} while (0)
+
+#define WM_TEST_LOG_START						\
+	do {										\
+		printf("-->%s\n", __FUNCTION__);		\
+	} while (0)
+
+
+#define WM_TEST_LOG_END							\
+	do {										\
+		printf("<--%s\n", __FUNCTION__);		\
+	} while (0)
+
+/*
+ * supported security method
+ */
+static const char *wifi_test_auth_method[] = {
+	"open",
+	"wep_shared",
+	"wpa_aes",
+	"wpa2_aes",
+	"wpa2_mixed",
+};
+
+
+static const wifi_manager_ap_auth_type_e auth_type_table[] = {
+	WIFI_MANAGER_AUTH_OPEN,                    /**<  open mode                      */
+	WIFI_MANAGER_AUTH_WEP_SHARED,              /**<  use shared key (wep key)       */
+	WIFI_MANAGER_AUTH_WPA_PSK,                 /**<  WPA_PSK mode                   */
+	WIFI_MANAGER_AUTH_WPA2_PSK,                /**<  WPA2_PSK mode                  */
+	WIFI_MANAGER_AUTH_WPA_AND_WPA2_PSK,        /**<  WPA_PSK and WPA_PSK mixed mode */
+	WIFI_MANAGER_AUTH_UNKNOWN,                 /**<  unknown type                   */
+};
+
+
+static const wifi_manager_ap_crypto_type_e crypto_type_table[] = {
+	WIFI_MANAGER_CRYPTO_NONE,                  /**<  none encryption                */
+	WIFI_MANAGER_CRYPTO_UNKNOWN,               /**<  to be fixed            */
+	WIFI_MANAGER_CRYPTO_AES,                   /**<  AES encryption                 */
+	WIFI_MANAGER_CRYPTO_AES,                   /**<  AES encryption                 */
```

Comment:
is it required to have repeated entries? are they mapped one-on-one to auth_types above?